rubicon/objc/runtime.py

```diff
@@ -671,7 +671,8 @@ def send_message(receiver, selector, *args, restype, argtypes, varargs=None):
             .format(tp=type(receiver))
         )
 
-    selector = SEL(selector)
+    if not isinstance(selector, SEL):
+        selector = SEL(selector)
```

Comment:
This sort of "universal coercion" makes me a little nervous; What's the use case for this change? Since we're doubling down on using `send_message`, is there any reason that we shouldn't go the other way here and enforce `if not isinstance(selector, SEL): raise TypeError()` (or similar)?

Reply:
To clarify: the previous code meant that `send_message` only accepted `str` and `bytes` objects as the `selector` parameter, and passing a real `SEL` object would cause a `TypeError`, even though the documentation said that this was allowed (see 18b2ccdf9c6d9cf35bf9d8622237b1ea983eea21).

In general I agree that argument type mismatches should be an error and we shouldn't try to automatically "fix" the type like this. But in this case there is a good use cases for passing `str`/`bytes` as the `selector`, namely calling `send_message` manually (`send_message(NSString, "string", restype=objc_id, argtypes=[])`). Also, passing `str`/`bytes` as `selector` has been supported for a long time (and until now has been the *only* way to call `send_message`) and is done often in practice, so I don't think we should forbid that.

Reply:
Ah - I was missing that this was really about the historical str/bytes usage. Agreed that is something that is both (a) needed for backwards compatibility, and (b) makes sense to preserve as an API (since it's an obvious usage)

Following the logic down the chain, it's going to break because it won't be castable to `bytes` as part of constructing the SEL, so I guess there's not a whole lot of risk here.